gulp/tasks/jekyll-build-preview.js

```diff
@@ -0,0 +1,29 @@
+(function() {
+    'use strict';
+
+    var gulp = require('gulp'),
+        gitUtils = require('../util/gitUtils'),
+        childProcess = require('child_process'),
+        previewConfigFile = '_tmp_preview_config.yml',
+        previewSiteDir = '_preview_site';
+
+    gulp.task('jekyll-build-preview', function() {
+        var branch = gitUtils.currentBranch(),
+            previewBaseUrl = '/' + branch + '/';
+        // Create a temporary Jekyll configuration file which specifies the base URL for the preview site
+        childProcess.execSync(
```

Comment:
Well FWIW, the commands could fail whether initiated from the a Makefile, shell script, python script, or gulp task.

I actually feel like surrounding this with a try/catch and console.logging something might help readability.